Clinical-Genomics/scout
# Filter variants by any gene panel from the case institute or the collaborators

fix #2006, fix #2005 and fix #1637.
This PR allows to filter variants of any type, from any case, by any existing gene panel in the database. 🕺 

**Current situation**:
- Variants filter accepts gene panels and gene symbols. You can specify both in the filter form.
- When you filter by gene panels, the only gene panels you can choose from are the default gene panels for the case.
- When the variants query is defined, if gene panels are provided, then the search returns variants with gene panel variant["gene_panels"]
- Variants page defaults to a default panel selected (panel1) or default panels for the case.

**Changes introduced by this PR:**
 - Variants filter still accepts gene panels and gene symbols, but all gene panels are translated into gene symbols before the query, and only gene symbols are used in variants query. (Simpler I think!)
- The initial variants filter contains the case-default panels + the eventual HPO panel.
- From the institute settings one can select any gene panel for the institute and the collaborator institutes (only the newest versions of each panel)
- Variants page still default to default panels for the case, but panel selection options will be broader whenever other institute panels are chosen.

**How to test -- Better on clinical-db stage**:
1. Install the branch on clinical-db stage
1. Try to go to the variants page of a case and check that the gene panels select contains case default panels + hpo panel.
1. Pre-select some different panels from the institute settings page.
1. Go back to the variants page and notice that all pre-selected panels show up, along with the case default panels and the HPO one 
1. Test the filtering, by gene panel(s), genes + gene panes, clinical filter, reset filter, whatever you come up with really.

Checklist when testing the functionality
- [x] Filter by empty OMIM panel should not return any results - same as now
- [x] Check and fix automatic filtering after loading of an existing filter
- [x] Check and fix clinical filter. This was fixed by #2044
- [x] Make sure that all default case panels are available in the panel select.
- [x] Automatic filtering by default panels when you click on the variants button
- [x] Only institute specific panels + panels of collaborators should be available to pre-select in the institute 
settings

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
